.evergreen/auth_oidc/setup_oidc.js

```diff
@@ -9,4 +9,7 @@ assert(admin.auth("bob", "pwd123"));
 admin.runCommand({createRole: 'test1/readWrite', roles:[{role: 'readWrite', db: 'test'}], privileges: []});
 admin.runCommand({createRole: 'test2/read', roles:[{role: 'read', db: 'test'}], privileges: []});
 
+// Wait for the node to be primary.
+while (admin.runCommand('replSetGetStatus').myState != 1) {}
```

Comment:
It's probably a good idea to add a small sleep to this loop to avoid flooding the server with commands.